exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java

```diff
@@ -82,17 +88,36 @@ public void createPublicApiHandlers(Node node, Router router) {
 
     return node.withSnapshot(view -> {
       CryptocurrencySchema schema = new CryptocurrencySchema(view);
-      ListIndex<HistoryEntity> history = schema.walletHistory(ownerKey);
-
-      List<HistoryEntity> result = new ArrayList<>();
-      for (HistoryEntity tx : history) {
-        result.add(tx);
-      }
-
-      return result;
+      ListIndex<HashCode> walletHistory = schema.transactionsHistory(ownerKey);
+      Blockchain blockchain = Blockchain.newInstance(view);
+      MapIndex<HashCode, TransactionMessage> txMessages = blockchain.getTxMessages();
```

Comment:
With no testkit it isn't possible to test these operations easily. I _could_ have put some data into `txMessages` in test setup code, but … we had decided on a readonly `Blockchain`, hadn't we?